diff --git a/bin/dune b/bin/dune
index d94b6ec2..abd02e00 100644
--- a/bin/dune
+++ b/bin/dune
@@ -4,6 +4,7 @@
  (preprocess
   (pps lwt_ppx ppx_let ppx_deriving_yojson ppx_deriving.std))
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)
    "-open"
diff --git a/esy-build-package/bin/dune b/esy-build-package/bin/dune
index 1c0b8479..74301b9d 100644
--- a/esy-build-package/bin/dune
+++ b/esy-build-package/bin/dune
@@ -3,6 +3,7 @@
  (public_name esyBuildPackageCommand)
  (modules esyBuildPackageCommand)
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)))
  (preprocess
@@ -15,6 +16,7 @@
  (public_name esyRewritePrefixCommand)
  (modules esyRewritePrefixCommand)
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)))
  (preprocess
diff --git a/esy.json b/esy.json
index 64d7da65..4722ee5d 100644
--- a/esy.json
+++ b/esy.json
@@ -32,6 +32,8 @@
     "test:e2e-slow": "node test-e2e-slow/run-slow-tests.js"
   },
   "resolutions": {
+    "ocaml": "4.10.1002-musl.static.flambda",
+    "@opam/astring": "github:dune-universe/astring#baa4e7e2bf919293aac2c899786c2ac40757dd9b",
     "@opam/bos": {
       "source": "github:esy-ocaml/bos:opam#a241814516fbe2",
       "override": {"dependencies": {"@opam/mtime": "*"}}
