diff --git a/bin/dune b/bin/dune
index d94b6ec2..abd02e00 100644
--- a/bin/dune
+++ b/bin/dune
@@ -4,6 +4,7 @@
  (preprocess
   (pps lwt_ppx ppx_let ppx_deriving_yojson ppx_deriving.std))
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)
    "-open"
diff --git a/esy-build-package/bin/dune b/esy-build-package/bin/dune
index 47148185..f9533d73 100644
--- a/esy-build-package/bin/dune
+++ b/esy-build-package/bin/dune
@@ -2,6 +2,7 @@
  (name esyBuildPackageCommand)
  (modules esyBuildPackageCommand)
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)))
  (preprocess
@@ -13,6 +14,7 @@
  (name esyRewritePrefixCommand)
  (modules esyRewritePrefixCommand)
  (flags
+  (-ccopt -static)
   (:standard
    (-w -39)))
  (preprocess
diff --git a/esy.json b/esy.json
index 29586322..c6e9634d 100644
--- a/esy.json
+++ b/esy.json
@@ -33,6 +33,7 @@
     "test:e2e-slow": "node test-e2e-slow/run-slow-tests.js"
   },
   "resolutions": {
+    "ocaml": "4.12.0-musl.static.flambda",
     "@opam/bos": "github:esy-ocaml/bos:opam#9c7956dda6c0d82afcd599c26f8f7efcc373b70e",
     "@opam/bigstringaf": "0.6.1",
     "@opam/cmdliner": "github:esy-ocaml/cmdliner:cmdliner.opam#e9316bc",
